src/ambianic/pipeline/avsource/av_element.py

```diff
@@ -104,6 +105,22 @@ def _fetch_img_exception_recovery(self):
         log.debug("Pausing for a moment to let remote network issues settle")
         time.sleep(1)
 
+    def _run_picamera_fetch(self):
+
+        picamera = Picamera()
+        if picamera.has_failure():
+            log.error("Picamera setup failed, exiting..")
+            return None
+
+        while not self._stop_requested:
+            try:
+                self.receive_next_sample(image=picamera.acquire())
```

Comment:
Does it make sense to add a few coverage tests for this section of code that codecov is complaining about?  The _TestPiCamera  mockup of the Picamera class that you already use for other unit tests should be still applicable here.

Reply:
There is a test for this in test_av_element but seems it is not working, will check again